src/stac_fastapi/geoparquet/api.py

```diff
@@ -0,0 +1,110 @@
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import Any, AsyncIterator, TypedDict
+
+import tomllib
+from fastapi import FastAPI
+from rustac import DuckdbClient
+
+import stac_fastapi.api.models
+from stac_fastapi.api.app import StacApi
+from stac_fastapi.extensions.core.pagination import OffsetPaginationExtension
+
+from .client import Client
+from .search import SearchGetRequest, SearchPostRequest
+from .settings import Settings, StacFastapiGeoparquetSettings
+
+GetSearchRequestModel = stac_fastapi.api.models.create_request_model(
+    model_name="SearchGetRequest",
+    base_model=SearchGetRequest,
+    mixins=[OffsetPaginationExtension().GET],
+    request_type="GET",
+)
+PostSearchRequestModel = stac_fastapi.api.models.create_request_model(
+    model_name="SearchPostRequest",
+    base_model=SearchPostRequest,
+    mixins=[OffsetPaginationExtension().POST],
+    request_type="POST",
+)
+
+
+class State(TypedDict):
+    """Application state."""
+
+    client: DuckdbClient
+    """The DuckDB client.
+    
+    It's just an in-memory DuckDB connection with the spatial extension enabled.
+    """
+
+    collections: dict[str, dict[str, Any]]
+    """A mapping of collection id to collection."""
+
+    hrefs: dict[str, str]
+    """A mapping of collection id to geoparquet href."""
+
+
+def create(
+    settings: Settings | None = None,
+    duckdb_client: DuckdbClient | None = None,
+) -> StacApi:
```

Comment:
I want to allow someone downstream to customize their `DuckdbClient` before building the app, e.g. if they want to enable some AWS credentials, set up some different extension locations, etc. In the function, I pass the `DuckdbClient` into the `FastAPI` app as an `extra`, then pull it out of extras and return it from the lifetime state. Seems to work, but might be a bit off-label in its usage?

Reply:
off-label 🤷  but I like it 👍

Reply:
Awesome, thanks for sanity-checking 🙇🏼 ❤️